vscode-extensions/tls-sync-vscode-shared/src/VScodeOutputChannelTerminalProvider.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { ITerminalProvider, TerminalProviderSeverity } from '@rushstack/terminal';
+import type * as vscode from 'vscode';
+
+/**
+ * Options to be provided to a {@link VScodeOutputChannelTerminalProvider}
+ *
+ * @beta
+ */
+export interface IVScodeOutputChannelTerminalProviderOptions {
+  /**
+   * If true, print verbose logging messages.
+   */
+  verboseEnabled: boolean;
+
+  /**
+   * If true, print debug logging messages. Note that "verbose" and "debug" are considered
+   * separate message filters; if you want debug to imply verbose, it is up to your
+   * application code to enforce that.
+   */
+  debugEnabled: boolean;
+}
+
+/**
+ * Terminal provider that prints to STDOUT (for log- and verbose-level messages) and
+ * STDERR (for warning- and error-level messages).
+ *
+ * @beta
+ */
+export class VScodeOutputChannelTerminalProvider implements ITerminalProvider {
+  private readonly _outputChannel: vscode.OutputChannel;
+  private _lineBuffer: string = '';
+  public static readonly supportsColor: boolean = false;
+
+  /**
+   * If true, verbose-level messages should be written to the console.
+   */
+  public verboseEnabled: boolean;
+
+  /**
+   * If true, debug-level messages should be written to the console.
+   */
+  public debugEnabled: boolean;
+
+  /**
+   * {@inheritDoc ITerminalProvider.supportsColor}
+   */
+  public readonly supportsColor: boolean = VScodeOutputChannelTerminalProvider.supportsColor;
+
+  public constructor(
+    outputChannel: vscode.OutputChannel,
+    options: Partial<IVScodeOutputChannelTerminalProviderOptions> = {}
+  ) {
+    this.verboseEnabled = !!options.verboseEnabled;
+    this.debugEnabled = !!options.debugEnabled;
+    this._outputChannel = outputChannel;
+  }
+
+  /**
+   * {@inheritDoc ITerminalProvider.write}
+   */
+  public write(data: string, severity: TerminalProviderSeverity): void {
+    // build _lineBuffer as data comes in
+    this._lineBuffer += data;
+    // when a newline is encountered, flush the _lineBuffer, log the line to outputChannel.appendLine
+    const lines: string[] = this._lineBuffer.split('\n');
+    this._lineBuffer = lines.pop() || ''; // keep the last line in the buffer if it doesn't end with a newline
+    this._outputChannel.appendLine(lines.join('\n') + '\n');
```

Comment:
```suggestion
    const lineBuffer: string = this._lineBuffer;
    const endIndex: number = lineBuffer.lastIndexOf('\n') + 1;
    if (endIndex > 0) {
      this._outputChannel.appendLine(lineBuffer.slice(0, endIndex);
    }
    // keep the last line in the buffer if it doesn't end with a newline
    this._lineBuffer = lineBuffer.slice(endIndex);
```

Reply:
This all feels very similar to this method?

https://github.com/microsoft/rushstack/blob/d825a62210ce587467374115c6851f83b3b12920/libraries/node-core-library/src/Text.ts#L63

May just want to reuse that or other methods in the `Text` class, or if none are exactly applicable, update the class to have a new/modified method that is

Reply:
Updated to use `Text.readLinesFromIterable`